site/src/pages/UserSettingsPage/TokensPage/hooks.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  useQuery,
+  useMutation,
+  useQueryClient,
+  QueryKey,
+} from "@tanstack/react-query"
+import { getTokens, deleteAPIKey } from "api/api"
+import { TokensFilter } from "api/typesGenerated"
+
+export const useTokensData = ({ include_all }: TokensFilter) => {
+  const queryKey = ["tokens", include_all]
+  const result = useQuery({
+    queryKey,
+    queryFn: () =>
+      getTokens({
+        include_all,
+      }),
+  })
+
+  return {
+    queryKey,
+    ...result,
+  }
+}
+
+export const useDeleteToken = (queryKey: QueryKey) => {
+  const queryClient = useQueryClient()
+
+  return useMutation({
+    mutationFn: deleteAPIKey,
+    onSuccess: () => {
+      // Invalidate and refetch
+      void queryClient.invalidateQueries(queryKey)
```

Comment:
Do we need `void` here?

Reply:
Good question. We have to add it to escape the following eslint error:
`Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator`.